### PR TITLE
feat: Add CI config

### PR DESCRIPTION
**Pull request by xiaoxli**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        stage("Checkout") {
+            checkout scm
+        }
+        stage("Install") {
+            sh 'rm -f package-lock.json'
+            sh 'npm install --no-package-lock'
+        }
+        stage("Lint") {
+            sh 'npm run lint -s'
+        }
```
Comment (SimonSchick) [inline Jenkinsfile]: What about tests? cc @saadataz

**Comment by connor4312** [comment]
@xiaoxli we use [Travis](https://travis-ci.org) for open source projects.

**Comment by xiaoxli** [comment]
I just removed the Jenkinsfile, and added a .travis.yml in the project. Needing some help.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+language: node_js
+node_js:
+  - "4"
```
Comment (xiaoxli) [inline .travis.yml]: Should I remove this version here? It's having different error than other versions.

[the remark this answers is not in the document]
Comment (xiaoxli) [reply .travis.yml]: I just removed 4 and 6.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+language: node_js
+node_js:
+  - "8"
+  - "9"
+install:
```
Comment (saadataz) [inline .travis.yml]: can you also add test stage here running `npm run test`

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+language: node_js
+node_js:
+  - "8"
+  - "9"
```
Comment (saadataz) [inline .travis.yml]: like Simon said, just 8 should be good here

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+language: node_js
+node_js:
+  - "8"
+install:
+  - npm install
```
Comment (SimonSchick) [inline .travis.yml]: Redundant, `npm install` is the default action for `node_js` language travis files.

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+  - npm run test
```
Comment (SimonSchick) [inline .travis.yml]: I don't think we can these, they require a browser.

File: .travis.yml [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+script:
+  - npm run build -s
```
Comment (SimonSchick) [inline .travis.yml]: Please add `npm run test:lint` here.

File: .travis.yml (hunk 6 above)
Comment (xiaoxli) [reply .travis.yml]: Removed.

Comment (saadataz) [reply .travis.yml]: Headless Chrome @SimonSchick ?
https://docs.travis-ci.com/user/gui-and-headless-browsers/#Using-the-Chrome-addon-in-the-headless-mode

Comment (xiaoxli) [reply .travis.yml]: I made the headless chrome browser work. Here are the references to make it work: https://developers.google.com/web/updates/2017/06/headless-karma-mocha-chai
https://github.com/travis-ci/travis-ci/issues/8836

File: .travis.yml (hunk 7 above)
Comment (xiaoxli) [reply .travis.yml]: I added headless chrome according to @saadataz comment.

File: .travis.yml (hunk 5 above)
Comment (xiaoxli) [reply .travis.yml]: Removed

File: .travis.yml (hunk 3 above)
Comment (xiaoxli) [reply .travis.yml]: Added after npm run build

File: .travis.yml (hunk 4 above)
Comment (xiaoxli) [reply .travis.yml]: Done

**Pull request merged**